formspree/forms/models.py

```diff
@@ -179,7 +179,8 @@ def send(self, data, keys, referrer):
 
         # sometimes we'll delete all archived submissions over the limit
         if random.random() < settings.EXPENSIVELY_WIPE_SUBMISSIONS_FREQUENCY:
-            records_to_keep = settings.ARCHIVED_SUBMISSIONS_LIMIT
+            records_to_keep = settings.ARCHIVED_SUBMISSIONS_LIMIT if self.upgraded \
+                    else settings.MONTHLY_SUBMISSIONS_LIMIT
```

Comment:
Ok so we actually want to keep the full archive submission limit for even non-upgraded forms. That way users can upgrade to get access to their last 1000 form submissions. Also if they've missed some submissions due to not being upgraded, we keep them around so they can get access once they upgrade.